File: DesignPatternsExercises/java/BuilderDesignPattern/Solution/ComputerBuilder.java
// public class ComputerBuilder __________(3)__________ PCBuilder {
public class ComputerBuilder extends PCBuilder {
    @Override
    public PCBuilder addMouse(String type) {
        // Mouse mouse = __________(4)__________(type);
        Mouse mouse = new Mouse(type);
        computer.addMouse(mouse);
        // __________(5)__________;
        return this;
    }
}

File: DesignPatternsExercises/ruby/builder_design_pattern/solution/computer_builder.rb
require_relative 'mouse'
require_relative 'pc_builder'

# class ComputerBuilder < __________(2)__________
class ComputerBuilder < PCBuilder
  def add_mouse(type)
    # mouse = __________(3)__________(type)
    mouse = Mouse.new(type)
    @computer.add_mouse mouse
    # __________(4)__________
    self
  end
end
